rewrite-maven/src/main/java/org/openrewrite/maven/internal/MavenPomDownloader.java

```diff
@@ -680,7 +680,7 @@ public Pom download(GroupArtifactVersion gav,
                             //If the exception is a common, client-side exception, cache an empty result.
                             mavenCache.putPom(resolvedGav, null);
                         }
-                    } catch (IOException e) {
+                    } catch (IOException | UncheckedIOException e) {
```

Comment:
`UncheckedIOException` can be thrown here by `RawPom.parse`:
https://github.com/openrewrite/rewrite/blob/ca705bcad67d2ef4ce86adbfce9ec4bc676273c1/rewrite-maven/src/main/java/org/openrewrite/maven/internal/RawPom.java#L153-L163

Or by OkHttpSender:
https://github.com/openrewrite/rewrite/blob/cdb945ff320f58af236f844c66a60c3d3582219f/rewrite-maven/src/test/java/org/openrewrite/maven/http/OkHttpSender.java#L72-L79

Or by HttpUrlConnectionSender:
https://github.com/openrewrite/rewrite/blob/52c770380b60955facd93db47d91a0aa2ae09462/rewrite-core/src/main/java/org/openrewrite/ipc/http/HttpUrlConnectionSender.java#L71-L126

Reply:
I did not find additional cases where we `catch(IOException ...)` and ourselves `throw new UncheckedIOException`. I'd think it's an improvement to now always catch those `UncheckedIOException`s as well, and treat them the same, such that we get the uniform `MavenDownloadingException` along with details needed to follow up, as opposed to the missing details reported in #5558.

Reply:
LGTM